blueprints/profile-param-compare-manage.md

```diff
@@ -0,0 +1,238 @@
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Profile Parameter Comparison / Management 
+
+https://github.com/apache/trafficcontrol/issues/2391
+
+## Problem Description
+<!--
+*What* is being asked for?
+*Why* is this necessary?
+*How* will this be used?
+-->
+As an operations user, I need the ability to easily compare parameters across 2 profiles and easily add or remove parameters from each profile. Currently in Traffic Portal, you can compare 2 profiles but it simply involves a comparison of the JSON returned from calls to GET /api/profiles/:profileid_1 and GET /api/profiles/:profileid_2. Any reconciliation of the 2 profiles would involve manually noting the differences and then navigating to either profile and making the necessary changes. This is not the best user experience. It is time-consuming and is prone to error.
+
+## Proposed Change
+<!--
+*How* will this be implemented (at a high level)?
+-->
+The current view in Traffic Portal (https://tp.domain.com/#!/profiles/compare/:profile_id_1/:profile_id_2) can be redesigned to list a superset of all parameters across the 2 profiles and show their profile membership side-by-side. It can also provide the ability to remove or add parameters to each profile easily and allow the user to persist the final state of both profiles (or reset to the original state). As the user makes changes, a visible indicater can be provided (i.e. red checkboxes) to show what has been changed.
+
+![alt text](img/profile-param-compare-manage.png "Mockup of proposed profile comparison view")
```

Comment:
nit: alt text on images should be meaningful. You can probably just re-use the title

Reply:
@ocket8888 - are you saying change it to this?

`![title](img/profile-param-compare-manage.png "Mockup of proposed profile comparison view")`

Reply:
uhh, maybe. I mean something like:

```markdown
![Mockup of proposed profile comparison view](img/profile-param-compare-manage.png "Mockup of proposed profile comparison view")
```

Reply:
weird. both ways work. /shrug

Reply:
btw i found the syntax here: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#images

Reply:
Nah, if you look at the source of the example images in that cheat sheet you'll see that they have literally `alt="alt text"` - title &#8800; alt text. Alt text is what's shown in user agents that don't support images and/or the image type of the image. It's an accessibility concern. Meanwhile:

> _"The `title` global attribute contains text representing advisory information related to the element it belongs to."_ - MDN

There's some other stuff including inheritance that differ between them, but the big difference is advisory information vs alternative representation.

Reply:
fixed